syntax = "proto3";

message ConfigChangeRequest {
  oneof ConfigType {
    EnqueueRule enqueue_rule = 1;
    RoutingRule routing_rule = 2;
    QueueInfo queue_info = 3;
  }
  string reporter = 4;
}

message EnqueueRule {
  message Change {
    string method = 1;
    string queue = 2;
    repeated string features = 3;
    int32 priority = 4; 
  }
  repeated Change changes = 1;
}

message RoutingRule {
  message Change {
  	string method = 1;
  	string queue = 2;
  	repeated string possible_routes = 3;
	}
  repeated Change changes = 4;
}

message QueueInfo {
	message Change {
  	string method = 1;
  	string queue = 2;
  	repeated string owner = 3;
	}
  repeated Change changes = 4;
}
